plaso/storage/redis/merge_reader.py

```diff
@@ -0,0 +1,285 @@
+    attribute containers are merged in the correct order.
+
+    Returns:
+      list[str]: names of the container types to merge.
+    """
+    container_types = []
+    for container_type in self._CONTAINER_TYPES:
+      # pylint: disable=protected-access
+      if self._store._HasAttributeContainers(container_type):
+        container_types.append(container_type)
+    return container_types
+
+  def _GetAttributeContainers(
+      self, container_type, callback=None, cursor=0, maximum_number_of_items=0):
+    """Retrieves attribute containers of the specified type.
+
+    Args:
+      container_type (str): attribute container type.
+      callback (function[StorageWriter, AttributeContainer]): function to call
+          after each attribute container is deserialized.
+      cursor (int): Redis cursor for scanning items.
+      maximum_number_of_items (Optional[int]): maximum number of
+          containers to retrieve, where 0 represent no limit.
+
+    Returns:
+      list(AttributeContainer): attribute containers from Redis.
+    """
+    if not cursor:
+      cursor = 0
+
+    cursor, items = self._store.GetSerializedAttributeContainers(
+        container_type, cursor, maximum_number_of_items)
+
+    containers = []
+    identifiers_to_delete = []
+    for identifier_bytes, serialized_container in iter(items.items()):
+      identifier_string = codecs.decode(identifier_bytes, 'utf-8')
+      identifier = identifiers.RedisKeyIdentifier(identifier_string)
+      identifiers_to_delete.append(identifier)
+
+      container = self._DeserializeAttributeContainer(
+          self._active_container_type, serialized_container)
+      container.SetIdentifier(identifier)
+
+      if callback:
+        callback(self._storage_writer, container)
+
+      containers.append(container)
+
+    self._store.RemoveAttributeContainers(container_type, identifiers_to_delete)
+
+    self._active_cursor = cursor
+    containers = self._active_extra_containers + containers
+
+    if maximum_number_of_items:
+      self._active_extra_containers = containers[maximum_number_of_items:]
+
+    return containers[:maximum_number_of_items]
+
+  def MergeAttributeContainers(
+      self, callback=None, maximum_number_of_containers=0):
+    """Reads attribute containers from a task store into the writer.
+
+    Args:
+      callback (function[StorageWriter, AttributeContainer]): function to call
+          after each attribute container is deserialized.
+      maximum_number_of_containers (Optional[int]): maximum number of
+          containers to merge, where 0 represent no limit.
+
+    Returns:
+      bool: True if the entire task storage file has been merged.
+
+    Raises:
+      RuntimeError: if the add method for the active attribute container
+          type is missing.
+    """
+    if not self._container_types:
+      self._container_types = self._GetContainerTypes()
+
+    number_of_containers = 0
+    while (self._active_cursor or
```

Comment:
Why the different line continuation (formatting) all of sudden? Please change this to fill a line as much as possible as the rest of the codebase.